app/settlement/process-settlement.js

```diff
@@ -1,11 +1,13 @@
 const { updateSettlementStatus } = require('./update-settlement-status')
+const { getSettlementFilter } = require('./get-settlement-filter')
 const { sendProcessingReturnEvent } = require('../event')
 
 const processSettlement = async (settlement) => {
   if (settlement.settled) {
-    const updated = await updateSettlementStatus(settlement)
-    if (updated) {
-      await sendProcessingReturnEvent(settlement)
+    const filter = getSettlementFilter(settlement)
```

Comment:
Creating a function to decide what the filter should be for matching payment requests.